lib/rules/no-inner-declarations.js

```diff
@@ -66,6 +76,22 @@ module.exports = {
 
     create(context) {
 
+        const sourceCode = context.sourceCode;
+        const programAst = sourceCode.ast;
+        const ecmaVersion = context.languageOptions.ecmaVersion || 6;
```

Comment:
I think we are really checking for undefined here?
```suggestion
        const ecmaVersion = context.languageOptions.ecmaVersion ?? 6;
```